contentreview-impl/impl/src/java/org/sakaiproject/contentreview/impl/turnitin/TurnitinReviewServiceImpl.java

```diff
@@ -3370,25 +3405,52 @@ public String getLTIAccess(String taskId, String contextId){
 		}
 		return ltiUrl;
 	}
+
+	public String getLTIReportAccess(ContentReviewItem item)
+	{
+		String ltiReportsUrl = null;
+		String contentId = item.getContentId();
+		String assignmentId = item.getTaskId();
+		String siteId = item.getSiteId();
+		try
+		{
+			String ltiReportsId = siteService.getSite(siteId).getProperties().getProperty("turnitin_reports_lti_id");
+			ContentResource resource = null;
+			resource = contentHostingService.getResource(contentId);
+			ResourceProperties resourceProperties = resource.getProperties();
+			String ltiResourceId = resourceProperties.getProperty("turnitin_id");
+			if (ltiResourceId == null)
+			{
+				// Fallback: link to assignment
+				return getLTIAccess(assignmentId, siteId);
+			}
+			ltiReportsUrl = "/access/basiclti/site/" + siteId + "/content:" + ltiReportsId + ",resource:" + ltiResourceId;
```

Comment:
The URL handling is in https://github.com/sakaicontrib/sakai-turnitin/pull/21/
Implementing a URL like /access/turnitin/ sounds like somewhat of a big undertaking (lti code is ugly!), and it would set a precedent for all future deep LTI integrations to do similar work.
I'd be in favour of continuing to use checks on the site_id field in BasicLTISecurityServiceImpl.getHttpAccess() (line 379 in sakai-turnitin PR-21). If new deep integrations are added in the future, we can define a sakai.property defining a list the deep integration LTI site ids.